cabal-testsuite/src/Test/Cabal/TestCode.hs

```diff
@@ -31,12 +35,23 @@ instance Exception TestCode
 #endif
 
 displayTestCode :: TestCode -> String
-displayTestCode TestCodeOk            = "OK"
-displayTestCode (TestCodeSkip msg)    = "SKIP " ++ msg
-displayTestCode (TestCodeKnownFail t) = "OK (known failure, see #" <> show t <> ")"
-displayTestCode TestCodeUnexpectedOk  = "FAIL (unexpected success)"
-displayTestCode TestCodeFail          = "FAIL"
+displayTestCode TestCodeOk               = "OK"
+displayTestCode (TestCodeSkip msg)       = "SKIP " ++ msg
+displayTestCode (TestCodeKnownFail t)    = "OK (known failure, see #" <> show t <> ")"
+displayTestCode (TestCodeUnexpectedOk t) = "FAIL (unexpected success, see #" <> show t <> ")"
+displayTestCode TestCodeFail             = "FAIL"
+displayTestCode (TestCodeFlakyFailed t)  = "FLAKY (FAIL, see #" <> show t <> ")"
+displayTestCode (TestCodeFlakyPassed t)  = "FLAKY (OK, see #" <> show t <> ")"
 
 isTestCodeSkip :: TestCode -> Bool
 isTestCodeSkip (TestCodeSkip _) = True
 isTestCodeSkip _                = False
+
+isTestCodeFlaky :: TestCode -> Maybe (Bool, Int)
```

Comment:
Should I use more concrete datatypes than `Bool`?

Reply:
Honestly, the tuple bothers me more.

Reply:
Yeah, please, let's have something more descriptive than `(Bool, Int)`. While on it, how about introducing an alias `type IssueID = Int`?

E.g.

```haskell
type TestPassed = Bool
type IssueID = Int
data FlakyStatus
  = NotFlaky
  | Flaky TestPassed IssueID

isTestCodeFlaky :: TestCode -> FlakyStatus
```
(haven't checked if `fourmolu` would approve this indentation)

I see they used `Maybe` below for a similar use case, but I'd say it's a code smell.

Reply:
I'm adding some more stuff I thought about this evening. Will push something later, I think it will look much better. Thanks both

Reply:
Pushed. The commits should show a comprehensive story about the changes.